reader_anonymous_ip.go

```diff
@@ -49,19 +49,23 @@ func (r *AnonymousIPReader) Lookup(ip net.IP) (*AnonymousIP, error) {
 	return result, nil
 }
 
-func NewAnonymousIPReader(buffer []byte) (*AnonymousIPReader, error) {
+func NewAnonymousIPReaderType(buffer []byte, expectedTypes ...string) (*AnonymousIPReader, error) {
 	reader, err := newReader(buffer)
 	if err != nil {
 		return nil, err
 	}
-	if reader.metadata.DatabaseType != "GeoIP2-Anonymous-IP" {
-		return nil, errors.New("wrong MaxMind DB Anonymous-IP type: " + reader.metadata.DatabaseType)
+	if !isExpectedDatabaseType(reader.metadata.DatabaseType, expectedTypes...) {
+		return nil, errors.New("wrong database Anonymous-IP type: " + reader.metadata.DatabaseType)
```

Comment:
Please revert error messages for compatibility reasons, `MaxMind DB` means format specification, not a trade mark.